src/scripts/components/SmartTableHeader.jsx

```diff
@@ -0,0 +1,61 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  Icon,
+} from '.';
+
+class SmartTableHeader extends React.Component {
+  state = {
+    sortKey: this.props.sortKey,
+    activeCaretClassName: 'u-text-body',
+    headers: this.props.headers,
```

Comment:
I'm wondering whether the state should be managed in the parent completely. It is considered an anti-pattern to initialize state from props since it creates multiple sources of truth for the component (props vs state). If all we're doing with this component is setting the class for display based on the sorted column then I think it might be risky to have that live independently of the logic that sorts the actual column data. Let me know if that makes sense.

Basically I'm envisioning the parent just passing down props that tell this component whether the header should display as sorted and in which direction. The onClick function would point to a method passed with props. Not sure if that is better or not. Totally open for discussion. Again my only concern is that this component has an independent state which is handled with different methods than the actual sorting of the data. I guess I also don't see how clicking this thing can sort the data if it is not sending anything to a parent function? I might be missing something, just let me know.